lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -208,11 +211,11 @@ public void OnNext(RuntimeStop runtimeStop)
                 }
                 catch (Exception e)
                 {
-                    Utilities.Diagnostics.Exceptions.CaughtAndThrow(new InvalidOperationException("Cannot stop evaluator properly", e), Level.Error, "Exception during shut down.", Logger);
+                    _evaluatorExitLogger.LogExit(false);
+                    Utilities.Diagnostics.Exceptions.CaughtAndThrow(
```

Comment:
Is EvaluatorExitLogger supposed to log the last line in evaluator, or it would want to replace current multiple unclear logs before evaluator exit? Here CaughtAndThrow will log again.